src/odemis/gui/comp/overlay/world.py

```diff
@@ -1231,46 +1314,350 @@ def draw(self, ctx, selected, shift=(0, 0), scale=1.0, canvas=None):
+            else:
+                dist = util.perpendicular_distance(self.v_start_pos, self.v_end_pos, vpos)
+                if dist < self.HOVER_MARGIN:
+                    return gui.HOVER_LINE
+
+        return gui.HOVER_NONE
+
+    def debug_edges(self, ctx):
+        super(LabelTool, self).debug_edges(ctx)
+
+        if self._edges and self._label.render_pos:
+            text_rect = self._edges_to_rect(self._edges['t_l'], self._edges['t_t'],
+                                            self._edges['t_r'], self._edges['t_b'])
+
+            ctx.set_line_width(0.5)
+            ctx.set_dash([])
+
+            ctx.set_source_rgba(0.6, 1, 0.6, 1)
+            ctx.rectangle(*text_rect)
+            ctx.stroke()
+
+    def draw(self, ctx, selected, shift=(0, 0), scale=1.0, canvas=None, font_size=None):
+        """ Draw a ruler and display the size in meters next to it. If the ruler is selected,
+        highlight it and make it thicker. A canvas is passed in case of print-ready export
+        and the ruler overlay draws on it """
+        super(LabelTool, self).draw(ctx, selected, shift=(0, 0), scale=1.0, canvas=None, font_size=None)
+
+        # If no valid selection is made, do nothing
+        if None in (self.p_start_pos, self.p_end_pos) or self.p_start_pos == self.p_end_pos:
+            return
+
+        # HACK: We take the opportunity that the canvas is redrawn to check if it has been shifted/rescaled.
+        # If so, we update the view positions (for the mouse interaction).
+        shiftscale = (shift, scale)
+        update_view = self.last_shiftscale != shiftscale and canvas is None
+
+        # In case a canvas is passed, the rulers should be drawn on this given canvas.
+        if canvas is None:
+            canvas = self.cnvs
+
+        offset = canvas.get_half_buffer_size()
+        b_start = canvas.phys_to_buffer(self.p_start_pos, offset)
+        b_end = canvas.phys_to_buffer(self.p_end_pos, offset)
+
+        if update_view:
+            self._update_view_positions(b_start, b_end)
+            self.last_shiftscale = shiftscale
+
+        # unit vector for physical coordinates
+        dx, dy = self.p_end_pos[0] - self.p_start_pos[0], self.p_end_pos[1] - self.p_start_pos[1]
+
+        # unit vector for buffer (pixel) coordinates
+        dpx, dpy = b_end[0] - b_start[0], b_end[1] - b_start[1]
+
+        phi = math.atan2(dx, dy) % (2 * math.pi)  # phi angle  of the label line in radians
+
+        # Find the label line length by calculating the Euclidean distance
+        pixel_length = math.hypot(dpx, dpy)  # label lime length in pixels
+
+        # text is placed horizontal
+        self._label.deg = math.degrees(math.pi)
+
+        # Draws a black background for the ruler
+        ctx.set_line_width(2)
+        ctx.set_source_rgba(0, 0, 0, 0.5)
+        ctx.move_to(*b_start)
+        ctx.line_to(*b_end)
+
+        # Ruler of 1 pixel width. Highlight the selected ruler and make it slightly thicker (2 pixels)
```

Comment:
That comment is not very clear. I'd just say "Make the line thicker and a different colour if it's selected"